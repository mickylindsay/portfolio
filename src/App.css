.App {
  background-color: #282c34;
  min-height: 100vh;
  font-size: calc(10px + 2vmin);
  color: white;
}

.content {
  padding: 1rem 1rem;
  font-size: .5em;
}
