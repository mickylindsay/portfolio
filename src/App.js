import React from 'react';
import './App.css';
import Header from './components/Header'
import Portfolio from './components/Portfolio'
import Personal from './components/Personal'

class App extends React.Component {
  state = {
    phase: true
  }

  flip = () => { this.setState({phase: !this.state.phase }) }

  render() {
    return (
      <div className="App">
        <Header flip={this.flip} phase={this.state.phase}/>
        <div className="content">
          {this.state.phase ? <Portfolio/> : <Personal/>}
        </div>
      </div>
    );
  }
}

export default App;
