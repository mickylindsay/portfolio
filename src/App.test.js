import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App component', () => {
  it('matches snapshot in portolio mode', () => {
    const wrapper = shallow(<App/>);
    wrapper.setState({phase: true});
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot in personal mode', () => {
    const wrapper = shallow(<App/>);
    wrapper.setState({phase: false});
    expect(wrapper).toMatchSnapshot();
  });

  it('renders personal after phase flip', () => {
    const wrapper = shallow(<App/>);
    wrapper.instance().flip();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders portfolio after second phase flip', () => {
    const wrapper = shallow(<App/>);
    wrapper.instance().flip();
    wrapper.instance().flip();
    expect(wrapper).toMatchSnapshot();
  });
});
