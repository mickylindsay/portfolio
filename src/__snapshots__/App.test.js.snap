// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot in personal mode 1`] = `
<div
  className="App"
>
  <Header
    flip={[Function]}
    phase={false}
  />
  <div
    className="content"
  >
    <Personal />
  </div>
</div>
`;

exports[`App component matches snapshot in portolio mode 1`] = `
<div
  className="App"
>
  <Header
    flip={[Function]}
    phase={true}
  />
  <div
    className="content"
  >
    <Portfolio />
  </div>
</div>
`;

exports[`App component renders personal after phase flip 1`] = `
<div
  className="App"
>
  <Header
    flip={[Function]}
    phase={false}
  />
  <div
    className="content"
  >
    <Personal />
  </div>
</div>
`;

exports[`App component renders portfolio after second phase flip 1`] = `
<div
  className="App"
>
  <Header
    flip={[Function]}
    phase={true}
  />
  <div
    className="content"
  >
    <Portfolio />
  </div>
</div>
`;
